doc/sphinx/source/inspection/index.rst

```diff
@@ -8,10 +8,76 @@ ability to create image formats. When visualization is enabled, a ``dot`` file
 is generated, along with a ``png``. The default can be adjusted by setting the 
 ``NGRAPH_VISUALIZE_TREE_OUTPUT_FORMAT`` flag to another format, like PDF. 
 
-Note: Large graphs are usually not legible with formats like PDF.  
+.. note:: Large graphs are usually not legible with formats like PDF.  
 
-Large graphs may require additional work to get into a human-readable format. We 
-have provided a script to convert the `most common default output`_, nGraph JSON,
+Large graphs may require additional work to get into a human-readable format.
+
+Manual adjustments with Graphviz
+================================
+
+As we are visualizing the graph, we can make some tweaks to the generated ``dot`` 
+file to make routing more tractable for Graphviz. Note that there are many possible 
+ways to go about this; what follows here is one way. 
+
+First trim edges that, intuitively speaking, have long "skip distance", which can 
+be useful for training graphs; those tend to have very long feed-forward edges where 
+intermediate values from ``fprop`` are stored for later reuse in the ``bprop`` phase:
+
+.. code-block:: raw
+   
+   Actual Graph Structure      Visualization
+    n0                             n0
+    | \                            |  \
+    n1 \                           n1  [to n50]
+    |   |                          |
+    n2  |                          n2
+    |   |                          |
+    n3  |                          n3
+    |   |                          |
+   ...  |                         ...  [from n0]
+    |  /                           |  /
+   n50                            n50
+
+
+Efficiently detecting a "long skip" is a bit tricky. We want to come up with a metric 
+that is reasonably fast to compute, but that does not result in cuts that will split 
+the graph itself into multiple components. The heuristic we are using for the jump 
+distance between ``n`` and ``m`` is the maximum difference in maximum path 
+length from ``n`` and ``m`` to any result node that is reachable from both 
+``n`` and ``m`` (or ``0``, if no such result node exists). 
+
+Formally:
+
+**Compute-Heights-Above-Each-Parameter:** ``N``
+
+.. literalinclude:: ../../../../../ngraph/src/ngraph/pass/visualize_tree.cpp
+   :language: cpp
+   :lines: 71-82
```

Comment:
These lines are likely to move around without notice, so we might want to copy and paste into the file, or make a separate copy of the source in the doc tree.

Reply:
:+1: